### PR TITLE
Forms project completed

### PR DESCRIPTION
**Pull request by mkezfarias**
Good afternon Mr. TSE,

We have completed the forms rails project and would like to request a review if you may.

Thank you in advanced.

Luis Puente / Carlos de Medeiros

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
+  <label for= "email"> Email </label>
+  <input id="email" name="user[email]" type="text">
+  <label for="username"> Username </label>  
+  <input id="username" name="user[username]" type="text">
+  <label for= "password"> Password </label>
+  <input id="password" name="user[password]" type="password">
+  <input name ="submit" type="submit" value= "Submit">
+</form>
+<%= form_tag('/users') do %>
```
Comment (Ispirett) [inline app/views/users/new.html.erb:25]: This form is throwing and error.

```
param is missing or the value is empty: user
user_params     params.require(:user).permit(:username,:password,:email)   end    end
-- | --



```

**Review by Ispirett — CHANGES_REQUESTED** [review]
# status changes: required :+1: 
Great work so far @mkezmkez  @camfilho  with this project.
There is one bug to address :bug:  please see below.

File: app/views/users/new.html.erb (hunk 1 above)
Comment (camfilho) [reply app/views/users/new.html.erb:25]: Hello there,

Could you tell us how to replicate the bug?

Comment (Ispirett) [reply app/views/users/new.html.erb:25]: I commented the others forms went to users/new and tried to create a new record .

**Comment by camfilho** [comment]
## For the new reviewer
Please uncomment line 8 on Users controller in order to avoid the above error

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <% end %>
+  </ul>
+</div>
+<% end %>
+
+
+<%= form_for @user do |f| %>
+  <%= f.text_field :username, placeholder: "Username" %>
+  <%= f.text_field :email, placeholder: "example@example.com" %>
+  <%= f.text_field :password %>
```
Comment (maelfosso) [inline app/views/users/edit.html.erb]: Here it should be a password field

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  
+  def new
+    @user = User.new
+  end
+  
+  def create
+#  @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Remove commented lines

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  <%= text_field_tag(:password) %>
+  <%= submit_tag('Submit') %>
+<% end %>
+-->
+
+<%= form_for @user do |f| %>
+<%= f.text_field :username, placeholder: "Username" %>
+<%= f.text_field :email, placeholder: "example@example.com" %>
+<%= f.text_field :password %>
+<%= f.submit "Submit" %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: For password, you should use a password field

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @mkezmkez  @camfilho 

Great work so far @mkezmkez @camfilho with this project.
Just a few things to changes to make it approve

Kindly read the comments below, make the corresponding changes and submit a new code review request.
Thanks.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: app/views/users/edit.html.erb (hunk 2 above)
Comment (samgaco) [inline app/views/users/edit.html.erb]: - [ ] Please, pass this into a password field as required by @maelfosso before

**Review by samgaco — APPROVED** [review]
Hi @camfilho and @mkezmkez :high_brightness: 

One requirement from the previous review is still not addressed, please correct it before merging. Since it is just a small fix I will approve the project, so after committing that change you can merge and continue :+1: 


---------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**